### DISCUSSION TITLE
Add the ability to designate specific project files to separate PCK files

### DISCUSSION THREAD

**Discussion by JohnGabrielUK** [godotengine/godot-proposals#10429] [Editor]
# The Problem

Imagine that a Godot game has been published on Steam, and the developer now wishes to add a DLC expansion with its own unique assets (music, voice lines, et cetera). Right now, the DLC content can just be included with the base game, and only made accessible if the game verifies through Steamworks that the user has the DLC in their account. This can be an acceptable solution, as long as the DLC content is not so large that it irritates users without the DLC, who find themselves downloading a large patch that does nothing unless they pay more money.

Ideally, the DLC's assets can be relegated to a separate PCK file, which is only downloaded when the DLC has been purchased, and loaded by the game at the developer's discretion.

At present, developers _can_ use the Export feature's filters to exclude files, alongside the "Export PCK/ZIP" option to do this manually, but it is a clunky solution.

# The Solution

In the FileExplorer window and/or the Import window, an option could be added along the lines of "Destination PCK". By default, this is the main PCK which is either exported alongside the binary, or embedded within it. However, there is also the option to "Manage Package Files...", which allows the developer to add/rename/remove additional PCK files.

It is, of course, the developer's responsibility to ensure that this new PCK is loaded into the game.

# Mock the Feature, presented by Dara Ó Briain

In this example, my DLC includes some unique music tracks that I don't want to ship with the base game. 

![image](https://github.com/user-attachments/assets/e3363615-fe09-4a11-9735-4b046f0fc391)

In the screenshot above, I have already created a new PCK and assigned the `music/dlc` folder to it.

Clicking the "Manage PCKs" button would bring up a window allowing me to create, rename, and delete PCKs. Naturally, the default PCK cannot be renamed or deleted.

**Comment by simonschreibt-code** [comment]
It's not a real solution, but you might like my new series about DLCs with Godot where I show how you export your project + all DLCs with 1 click:

https://www.youtube.com/playlist?list=PLyVOyWtfqDg8__bs-tmyPy-F6EoOmXVDa

**Comment by JohnGabrielUK** [comment]
I didn't realise that Godot allows for multiple presets with the same export platform. So I could make a second export that excludes everything but the folder with the DLC in it, and have the primary export exclude just the DLC folder. It requires you to have your project structured a certain way, but it's less of a problem than I at first thought.
